Automattic/simplenote-android
# Implements In App Account Deletion

Fixes #1415 

### Fix
Implement a feature that let users delete their account from within the app. This feature sends a request to the Simplenote server which sends an email to the user. The user can then delete the account using the email with the required information. If the request fails because of Internet connection or server issues, then we show an error message to the user, otherwise, we show a received request message. The videos below show both flows:

**Light Mode**
| Successful Flow      | Error Flow |
| ----------- | ----------- |
| ![20210716_001406](https://user-images.githubusercontent.com/195721/125895508-44d36a29-ac0b-4366-bf65-fd45f3ae01c4.gif) | ![20210716_001512](https://user-images.githubusercontent.com/195721/125895527-7213c035-64db-4e98-909e-9e65e6a1985e.gif)      |


**Dark Mode**
| Successful Flow      | Error Flow |
| ----------- | ----------- |
| ![20210716_001948](https://user-images.githubusercontent.com/195721/125895958-8e8ea1a4-e87b-4aa6-ab63-0e3bad6d89f8.gif)  |  ![20210716_002042](https://user-images.githubusercontent.com/195721/125895979-1c159760-04a1-4dde-bf66-6c0b6c851854.gif)    |


More details about the project: p2XJRt-3bK-p2
Last Iteration of designs: p2XJRt-3cd-p2


### Test

**Note: use Simperium key for production and create accounts for tests. You can create accounts with the following format: youremail+n@automattic.com where n is an integer number (e.g. danilo.dominguez+1@automattic.com). Follow the instructions in the email to setup password.**

1. Open the left menu by tapping on the top-left corner icon
2. Tap on Settings
3. Scroll down and Tap on the preference `Delete Account`
4. That will open a dialog. Tap on `DELETE ACCOUNT` button
5. Open your email and proceed to delete account
6. Open Simplenote and it should logout automatically


### Review

This PR can be reviewed by @ParaskP7  or @hichamboushaba.

### Release
`RELEASE-NOTES.txt` was updated in fd9ac10108a556a42ce1b2dbe33c3babb63d90d5 with:
> Added In App Account Deletion
